internal/filters/filters.go

```diff
@@ -0,0 +1,159 @@
+		typeFilters []Filter
+	)
+	if typeFilters, ok = f.byType[typ]; !ok {
+		return false
+	}
+	for _, filter := range typeFilters {
+		if filter.Name == name {
+			return true
+		}
+	}
+	return false
+}
+
+// Filter contains a type + name (type = elasticsearch, name = my-cluster)
+// and a label selector to easily determine if any queried resource's labels match
+// a given filter.
+type Filter struct {
+	Type     string
+	Name     string
+	Selector labels.Selector
+}
+
+// New returns a new set of filters, given a slice of key=value pairs,
+// parses and validates the given filters, and returns an error
+// if the given key=value pairs are invalid.
+//
+// source example:
+// []string{"elasticsearch=my-cluster", "kibana=my-kb"}
+func New(source []string) (Filters, error) {
+	return parse(source)
+}
+
+// parse will validate the given source filters, and for each
+// filter type, will create a Filter with a label selector,
+// returning the set of Filters, and any errors encountered.
+func parse(source []string) (Filters, error) {
+	filters := Filters{
+		byType: map[string][]Filter{},
+	}
+	if len(source) == 0 {
+		return filters, nil
+	}
+	var typ, name string
+	for _, fltr := range source {
+		filterSlice := strings.Split(fltr, "=")
+		if len(filterSlice) != 2 {
+			return filters, fmt.Errorf("invalid filter: %s", fltr)
+		}
+		typ, name = filterSlice[0], filterSlice[1]
+		if err := validateType(typ); err != nil {
+			return filters, err
+		}
+		if _, ok := filters.byType[typ]; ok {
+			return filters, fmt.Errorf("invalid filter: %s: multiple filters for the same type (%s) are not supported", fltr, typ)
+		}
+		selector, err := processSelector(typ, name)
+		if err != nil {
+			return filters, fmt.Errorf("while processing label selector: %w", err)
+		}
+		filters.byType[typ] = append(filters.byType[typ], Filter{
+			Name:     name,
+			Type:     typ,
+			Selector: selector,
+		})
+	}
+	return filters, nil
+}
+
+func validateType(typ string) error {
+	if !slices.Contains(ValidTypes, typ) {
+		return fmt.Errorf("invalid filter type: %s, supported types: %v", typ, ValidTypes)
+	}
+	return nil
+}
+
+// processSelector will create a label set for a given type+name pair,
+// create a selector from that label set, and add label requirements
+// to the selector for each key/value in the label set, returning the selector
+// and any errors in processing.
+//
+// Having a labels.Selector on the Filter allows an easy match operation:
+//
+//	Selector.Match(LabelSet)
+//
+// against any runtime object's labels.
+func processSelector(typ, name string) (labels.Selector, error) {
+	nameAttr := "name"
+	if typ == "elasticsearch" {
+		nameAttr = "cluster-name"
+	}
+	set := labels.Set{
+		"common.k8s.elastic.co/type":                       typ,
+		fmt.Sprintf("%s.k8s.elastic.co/%s", typ, nameAttr): name,
+	}
+	s := labels.SelectorFromValidatedSet(set)
+	for sk, v := range set {
+		req, err := labels.NewRequirement(sk, selection.Equals, []string{v})
```

Comment:
`SelectorFromValidatedSet` already adds the Requirement, that's why I recommended using it.

Reply:
Going and looking at the internal implementation, I see this now.  Thank you.  Interesting that `labels.SelectorFromSet(set)` uses `SelectorFromValidatedSet` internally, so is functionally equivalent.